soroban-sdk-macros/src/lib.rs

```diff
@@ -346,11 +346,10 @@ pub fn contractmeta(metadata: TokenStream) -> TokenStream {
 
         let ident = format_ident!(
             "__CONTRACT_KEY_{}",
-            args.key
-                .as_bytes()
-                .iter()
-                .map(|b| format!("{b:02x}"))
-                .collect::<String>()
+            args.key.as_bytes().iter().fold(String::new(), |mut s, b| {
+                let _ = write!(s, "{b:02x}");
+                s
+            })
```

Comment:
I almost didn't make this change but clippy made the case that the original code creates a lot of strings and allocates more. It doesn't really matter though as this is a short bytes to convert. I mostly updated it to get rid of the clippy warning.